Snabble/UI/Scanner/ShoppingListCellView.swift

```diff
@@ -0,0 +1,161 @@
+        nameLabel.translatesAutoresizingMaskIntoConstraints = false
+        nameLabel.font = UIFont.systemFont(ofSize: 15)
+        nameLabel.textColor = .label
+        nameLabel.textAlignment = .natural
+        nameLabel.numberOfLines = 1
+
+        let tickLayoutGuide = UILayoutGuide()
+
+        let checkContainer = UIView()
+        checkContainer.translatesAutoresizingMaskIntoConstraints = false
+        checkContainer.backgroundColor = .systemTeal
+        checkContainer.layer.shadowColor = UIColor.gray.cgColor
+        checkContainer.layer.shadowOpacity = 0.3
+        checkContainer.layer.shadowOffset = CGSize.zero
+        checkContainer.layer.shadowRadius = 1.5
+        checkContainer.layer.masksToBounds = true
+        checkContainer.layer.borderWidth = 1 / UIScreen.main.scale
+        checkContainer.layer.borderColor = UIColor.systemBackground.cgColor
+        checkContainer.layer.cornerRadius = 12
+        checkContainer.backgroundColor = .clear
+
+        let checkImage = UIImageView()
+        checkImage.translatesAutoresizingMaskIntoConstraints = false
+        checkImage.contentMode = .scaleAspectFit
+        checkImage.image = Asset.SnabbleSDK.iconCheckLarge.image
+
+        let quantityLabel = UILabel()
+        quantityLabel.translatesAutoresizingMaskIntoConstraints = false
+        quantityLabel.font = UIFont.systemFont(ofSize: 13)
+        quantityLabel.textColor = .label
+        quantityLabel.textAlignment = .center
+        quantityLabel.numberOfLines = 1
+
+        addLayoutGuide(imageViewLayuotGuide)
+        addLayoutGuide(tickLayoutGuide)
+
+        addSubview(productImage)
+        addSubview(spinner)
+        addSubview(nameLabel)
+        addSubview(quantityLabel)
+        addSubview(checkContainer)
+        checkContainer.addSubview(checkImage)
+
+        self.productImage = productImage
+        self.spinner = spinner
+        self.nameLabel = nameLabel
+        self.quantityLabel = quantityLabel
+        self.checkContainer = checkContainer
+        self.checkImage = checkImage
+
+        NSLayoutConstraint.activate([
+            imageViewLayuotGuide.leadingAnchor.constraint(equalTo: leadingAnchor),
+            imageViewLayuotGuide.widthAnchor.constraint(equalToConstant: 42),
+            imageViewLayuotGuide.topAnchor.constraint(equalTo: topAnchor),
+            imageViewLayuotGuide.bottomAnchor.constraint(equalTo: bottomAnchor),
+
+            productImage.leadingAnchor.constraint(equalTo: imageViewLayuotGuide.leadingAnchor, constant: 14),
+            productImage.widthAnchor.constraint(equalToConstant: 28),
+            productImage.heightAnchor.constraint(equalToConstant: 28),
+            productImage.centerYAnchor.constraint(equalTo: imageViewLayuotGuide.centerYAnchor),
+
+            spinner.leadingAnchor.constraint(equalTo: imageViewLayuotGuide.leadingAnchor),
+            spinner.widthAnchor.constraint(equalToConstant: 28),
+            spinner.heightAnchor.constraint(equalToConstant: 28),
```

Comment:
The Activity Indicator should be centered in the box. It's pinned to the left at the moment. Please also add contraints to each boundary if you center it.

Reply:
Fixed: `spinner` centered by both axis related to `productImage`

Reply:
1. That's good, but could you add constraints to all four edges, that is never if an image is too small is drawn out of bounds?
2. Do we need the height and width constraint for the spinner? I don't think so cause it' has an `intrinsicContentSize`.